logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -101,10 +104,14 @@ private void addBody(final HttpMessage request, final ImmutableMap.Builder<Strin
         final String body = request.getBodyAsString();
 
         if (isJson(request.getContentType())) {
-            builder.put("body", body.isEmpty() ? JsonBody.EMPTY : new JsonBody(compactJson(body)));
-        } else {
-            addUnless(builder, "body", request.getBodyAsString(), String::isEmpty);
+            try {
+                builder.put("body", body.isEmpty() ? JsonBody.EMPTY : new JsonBody(compactJson(body)));
+                return;
```

Comment:
> Control flow is a bit confusing. Can we maybe extract a method returning an optional or nullable JsonBody?

Would it make sense to use some kind of SPI here to allow *embedding* bodies as json for different content types? Because with the current approach one always needs to change the `JsonHttpLogFormatter`.